agent/agentbootstrap/bootstrap.go

```diff
@@ -279,6 +284,54 @@ func verifyModelConfigDefaultSpace(st *state.State) error {
 	return errors.Annotatef(err, "cannot verify %s", config.DefaultSpace)
 }
 
+func validateCAASImageRepo(cfg controller.Config) error {
+	path := cfg.CAASImageRepo()
+	deprecated := cfg.CAASOperatorImagePath()
+
+	if path == "" && deprecated == "" {
+		return nil
+	}
+
+	// Ensure that if both are set, they are the same. We can't have divergent
+	// image repos. This is a temporary measure to allow for a transition period
+	// where we support both the old and new config keys.
+	if path != "" && deprecated != "" && path != deprecated {
+		return errors.NotValidf("CAASImageRepo and CAASOperatorImagePath cannot be different, if both set")
+	}
+
+	repoImage := deprecated
+	if repoImage == "" {
+		repoImage = path
+	}
+
+	imageRepo, err := imagerepo.NewImageRepo(repoImage)
+	if err != nil {
+		return errors.Annotatef(err, "invalid image repo %q", repoImage)
+	}
+
+	details, err := imageRepo.RequestDetails()
+	if err != nil {
+		return errors.Annotatef(err, "cannot get image repo details %q", repoImage)
+	}
+
+	// Nothing to set.
+	// TODO (stickupkid): I'm unsure what to actually do here, because in theory
+	// we need to remove the CAASImageRepo and CAASOperatorImagePath keys from
+	// the controller config.
+	if details.Empty() {
+		return nil
```

Comment:
I think in reality we should actually just return an error that we couldn't find the details we want and tell the operator about the problem.